Compilation warnings for src/gvars.c

I have the following warnings with current master: 
````
   C       src/gvars.c => obj/gvars.lo
src/gvars.c: In function ‘GVarName’:
src/gvars.c:577:10: warning: unused variable ‘oldPos’ [-Wunused-variable]
     UInt oldPos = pos;
          ^
src/gvars.c: At top level:
cc1: warning: unrecognized command line option "-Wno-unknown-warning-option"
cc1: warning: unrecognized command line option "-Wno-implicit-fallthrough"
````
Did not have them a couple days earlier, so I guess some new commit must be responsible. The version for cc is
````
ghorvath@DesktopPC:~/work/gap$ cc -v
Using built-in specs.
COLLECT_GCC=cc
COLLECT_LTO_WRAPPER=/usr/lib/gcc/x86_64-linux-gnu/4.9/lto-wrapper
Target: x86_64-linux-gnu
Configured with: ../src/configure -v --with-pkgversion='Debian 4.9.2-10' --with-bugurl=file:///usr/share/doc/gcc-4.9/README.Bugs --enable-languages=c,c++,java,go,d,fortran,objc,obj-c++ --prefix=/usr --program-suffix=-4.9 --enable-shared --enable-linker-build-id --libexecdir=/usr/lib --without-included-gettext --enable-threads=posix --with-gxx-include-dir=/usr/include/c++/4.9 --libdir=/usr/lib --enable-nls --with-sysroot=/ --enable-clocale=gnu --enable-libstdcxx-debug --enable-libstdcxx-time=yes --enable-gnu-unique-object --disable-vtable-verify --enable-plugin --with-system-zlib --disable-browser-plugin --enable-java-awt=gtk --enable-gtk-cairo --with-java-home=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64/jre --enable-java-home --with-jvm-root-dir=/usr/lib/jvm/java-1.5.0-gcj-4.9-amd64 --with-jvm-jar-dir=/usr/lib/jvm-exports/java-1.5.0-gcj-4.9-amd64 --with-arch-directory=amd64 --with-ecj-jar=/usr/share/java/eclipse-ecj.jar --enable-objc-gc --enable-multiarch --with-arch-32=i586 --with-abi=m64 --with-multilib-list=m32,m64,mx32 --enable-multilib --with-tune=generic --enable-checking=release --build=x86_64-linux-gnu --host=x86_64-linux-gnu --target=x86_64-linux-gnu
Thread model: posix
gcc version 4.9.2 (Debian 4.9.2-10) 
````

--- a/src/gvars.c
+++ b/src/gvars.c
@@ -531,6 +531,16 @@ Obj FuncGET_NAMESPACE(Obj self)
     return STATE(CurrNamespace);
 }
 
+
+static inline UInt HashString( const Char * name )
+{
+    UInt hash = 0;
+    while ( *name ) {
+        hash = 65599 * hash + *name++;
+    }
+    return hash;
+}
+
 /****************************************************************************
 **
 *F  GVarName(<name>)  . . . . . . . . . . . . . .  global variable for a name
@@ -548,7 +558,6 @@ UInt GVarName (
     Obj                 string;         /* temporary string value <name>   */
     Obj                 table;          /* temporary copy of <TableGVars>  */
     Obj                 gvar2;          /* one element of <table>          */
-    const Char *        p;              /* loop variable                   */
     UInt                i;              /* loop variable                   */
     Int                 len;            /* length of name                  */
 
@@ -564,17 +573,13 @@ UInt GVarName (
     }
 
     /* start looking in the table at the following hash position           */
-    pos = 0;
-    for ( p = name; *p != '\0'; p++ ) {
-        pos = 65599 * pos + *p;
-    }
+    const UInt hash = HashString( name );
 #ifdef HPCGAP
     LockGVars(0);
 #endif
-    pos = (pos % SizeGVars) + 1;
 
     /* look through the table until we find a free slot or the global      */
-    UInt oldPos = pos;
+    pos = (hash % SizeGVars) + 1;
     while ( (gvar = ELM_PLIST( TableGVars, pos )) != 0
          && strncmp( NameGVar( INT_INTOBJ(gvar) ), name, 1023 ) ) {
         pos = (pos % SizeGVars) + 1;
@@ -587,7 +592,7 @@ UInt GVarName (
         LockGVars(1);
 
         /* look through the table until we find a free slot or the global  */
-        pos = oldPos;
+        pos = (hash % SizeGVars) + 1;
         while ( (gvar = ELM_PLIST( TableGVars, pos )) != 0
              && strncmp( NameGVar( INT_INTOBJ(gvar) ), name, 1023 ) ) {
             pos = (pos % SizeGVars) + 1;
@@ -657,10 +662,7 @@ UInt GVarName (
             for ( i = 1; i <= (SizeGVars-1)/2; i++ ) {
                 gvar2 = ELM_PLIST( table, i );
                 if ( gvar2 == 0 )  continue;
-                pos = 0;
-                for ( p = NameGVar( INT_INTOBJ(gvar2) ); *p != '\0'; p++ ) {
-                    pos = 65599 * pos + *p;
-                }
+                pos = HashString( NameGVar( INT_INTOBJ(gvar2) ) );
                 pos = (pos % SizeGVars) + 1;
                 while ( ELM_PLIST( TableGVars, pos ) != 0 ) {
                     pos = (pos % SizeGVars) + 1;

--- a/src/records.c
+++ b/src/records.c
@@ -105,6 +105,15 @@ static inline void HPC_UnlockNames() {}
 #endif
 
 
+static inline UInt HashString( const Char * name )
+{
+    UInt hash = 0;
+    while ( *name ) {
+        hash = 65599 * hash + *name++;
+    }
+    return hash;
+}
+
 /****************************************************************************
 **
 *F  RNamName(<name>)  . . . . . . . . . . . . convert a name to a record name
@@ -121,31 +130,25 @@ UInt            RNamName (
 {
     Obj                 rnam;           /* record name (as imm intobj)     */
     UInt                pos;            /* hash position                   */
-    UInt                len;            /* length of name                  */
     Char                namx [1024];    /* temporary copy of <name>        */
     Obj                 string;         /* temporary string object <name>  */
     Obj                 table;          /* temporary copy of <HashRNam>    */
     Obj                 rnam2;          /* one element of <table>          */
-    const Char *        p;              /* loop variable                   */
     UInt                i;              /* loop variable                   */
 
-    /* start looking in the table at the following hash position           */
-    pos = 0;
-    len = 0;
-    for ( p = name; *p != '\0'; p++ ) {
-        pos = 65599 * pos + *p;
-        len++;
-    }
-    pos = (pos % SizeRNam) + 1;
-
-    HPC_LockNames(0); /* try a read lock first */
-    if (len >= 1023) {
+    if (strlen(name) >= 1023) {
         // Note: We can't pass 'name' here, as it might get moved by garbage collection
-        HPC_UnlockNames();
         ErrorQuit("Record names must consist of less than 1023 characters", 0, 0);
         return 0;
     }
+
+    /* start looking in the table at the following hash position           */
+    const UInt hash = HashString( name );
+
+    HPC_LockNames(0); /* try a read lock first */
+
     /* look through the table until we find a free slot or the global      */
+    pos = (hash % SizeRNam) + 1;
     while ( (rnam = ELM_PLIST( HashRNam, pos )) != 0
          && strncmp( NAME_RNAM( INT_INTOBJ(rnam) ), name, 1023 ) ) {
         pos = (pos % SizeRNam) + 1;
@@ -159,9 +162,10 @@ UInt            RNamName (
       HPC_UnlockNames(); /* switch to a write lock */
       HPC_LockNames(1);
       /* look through the table until we find a free slot or the global      */
+      pos = (hash % SizeRNam) + 1;
       while ( (rnam = ELM_PLIST( HashRNam, pos )) != 0
-	   && strncmp( NAME_RNAM( INT_INTOBJ(rnam) ), name, 1023 ) ) {
-	  pos = (pos % SizeRNam) + 1;
+           && strncmp( NAME_RNAM( INT_INTOBJ(rnam) ), name, 1023 ) ) {
+          pos = (pos % SizeRNam) + 1;
       }
     }
     if (rnam != 0) {
@@ -182,7 +186,7 @@ UInt            RNamName (
     SET_ELM_PLIST( NamesRNam,   CountRNam, string );
     CHANGED_BAG(   NamesRNam );
 
-    /* if the table is too crowed, make a larger one, rehash the names     */
+    /* if the table is too crowded, make a larger one, rehash the names     */
     if ( SizeRNam < 3 * CountRNam / 2 ) {
         table = HashRNam;
         SizeRNam = 2 * SizeRNam + 1;
@@ -197,10 +201,7 @@ UInt            RNamName (
         for ( i = 1; i <= (SizeRNam-1)/2; i++ ) {
             rnam2 = ELM_PLIST( table, i );
             if ( rnam2 == 0 )  continue;
-            pos = 0;
-            for ( p = NAME_RNAM( INT_INTOBJ(rnam2) ); *p != '\0'; p++ ) {
-                pos = 65599 * pos + *p;
-            }
+            pos = HashString( NAME_RNAM( INT_INTOBJ(rnam2) ) );
             pos = (pos % SizeRNam) + 1;
             while ( ELM_PLIST( HashRNam, pos ) != 0 ) {
                 pos = (pos % SizeRNam) + 1;

